src/async.rs

```diff
@@ -363,6 +362,50 @@ impl<S: Sleeper> AsyncClient<S> {
         self.post_request_hex("/tx", transaction).await
     }
 
+	/// Broadcast a package of [`Transaction`] to Esplora
+	///
+	/// if `maxfeerate` is provided, any transaction whose
+	/// fee is higher will be rejected
+	///
+	/// if  `maxburnamount` is provided, any transaction
+	/// with higher provably unspendable outputs amount
+	/// will be rejected
+	pub async fn broadcast_pkg(
+		&self,
+		transactions: &[Transaction],
+		maxfeerate: Option<f64>,
+		maxburnamount: Option<f64>)
+	-> Result<SubmitPackageResult, Error> {
+		let url = format!("{}/txs/package", self.url);
+
+		let serialized_txs = transactions
+			.iter()
+			.map(|tx| serialize(&tx).to_lower_hex_string())
+			.collect::<Vec<_>>();
+
+		let mut request = self.client.post(url).body(
+			serde_json::to_string(&serialized_txs).unwrap().as_bytes().to_vec()
+		);
+
+		if let Some(maxfeerate) = maxfeerate {
+			request = request.query(&[("maxfeerate", maxfeerate.to_string())])
+		}
+
+		if let Some(maxburnamount) = maxburnamount {
+			request = request.query(&[("maxburnamount", maxburnamount.to_string())])
+		}
+
```

Comment:
To my understanding the `maxreerate` and `maxburnamount` are a protection measure. It should protect the user against accidentally wasting all their funds.

I'd prefer to do make the checks locally before sending the request. There is are some varieties in `electrs` implementation and I don't want users to get surprised